modules/rostests/apitests/gdi32/StretchBlt.c

```diff
@@ -18,91 +18,852 @@
+        expected[1] = 0x00FF0000;
+        expected[2] = 0x0000FF00;
+        expected[3] = 0x000000FF;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                            1, 1, -2, -2, 0, 0, 2, 2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0011 (3) - Both Flip.
+
+    if ((SrcTopDown || DstTopDown) && !(SrcTopDown && DstTopDown))
+    {
+        expected[0] = 0xFF000000;
+        expected[1] = 0x00FF0000;
+        expected[2] = 0x0000FF00;
+        expected[3] = 0x000000FF;
+    }
+    else
+    {
+        expected[0] = 0x0000FF00;
+        expected[1] = 0x000000FF;
+        expected[2] = 0xFF000000;
+        expected[3] = 0x00FF0000;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                            1, 1, -2, -2, 0, 1, 2, -2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0010 (2) - Horizontal Flip. 
+
+    if ((SrcTopDown || DstTopDown) && !(SrcTopDown && DstTopDown))
+    {
+        expected[0] = 0x000000FF;
+        expected[1] = 0x0000FF00;
+        expected[2] = 0x00FF0000;
+        expected[3] = 0xFF000000;
+    }
+    else
+    {
+        expected[0] = 0x00FF0000;
+        expected[1] = 0xFF000000;
+        expected[2] = 0x000000FF;
+        expected[3] = 0x0000FF00;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                            1, 1, -2, -2, 1, 0, -2, 2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0001 (1) - Vertical Flip. 
+
+    if ((SrcTopDown || DstTopDown) && !(SrcTopDown && DstTopDown))
+    {
+        expected[0] = 0x00FF0000;
+        expected[1] = 0xFF000000;
+        expected[2] = 0x000000FF;
+        expected[3] = 0x0000FF00;
+
+    }
+    else
+    {
+        expected[0] = 0x000000FF;
+        expected[1] = 0x0000FF00;
+        expected[2] = 0x00FF0000;
+        expected[3] = 0xFF000000;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                            2, 2, -2, -2, 2, 2, -2, -2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0000 (0) - No Flip. Special Case.
+
+    SelectObject(hdcSrc, oldSrc);
+    DeleteObject(bmpSrc);
+    DeleteDC(hdcSrc);
+
+    SelectObject(hdcDst, oldDst);
+    DeleteObject(bmpDst);
+    DeleteDC(hdcDst);
+
+    DeleteDC(hdcScreen);
+}
 
 START_TEST(StretchBlt)
 {
+    trace("\n\r\nStart of Generalized StretchBlt tests.\n\n");
     test_StretchBlt();
+
+    trace("\n\r\nStart of Source Top Down and Destination Top Down tests.\n\n");
+    test_StretchBlt_TopDownOptions(TRUE, TRUE);
+
+    trace("\n\r\nStart of Source Top Down and Destination Bottom Up tests.\n\n");
+    test_StretchBlt_TopDownOptions(TRUE, FALSE);
+
+    trace("\n\r\nStart of Source Bottom Up and Destination Top Down tests.\n\n");
+    test_StretchBlt_TopDownOptions(FALSE, TRUE);
+
+    trace("\n\r\nStart of Source Bottom Up and Destination Bottom Up tests.\n\n");
+    test_StretchBlt_TopDownOptions(FALSE, FALSE);
```

Comment:
`"\r"` is not needed here: Is it something marking?

```suggestion
    trace("\n\n## Start of generalized StretchBlt tests.\n\n");
    test_StretchBlt();

    trace("\n\n## Start of source top-down and destination top-down tests.\n\n");
    test_StretchBlt_TopDownOptions(TRUE, TRUE);

    trace("\n\n## Start of source top-down and destination bottom-up tests.\n\n");
    test_StretchBlt_TopDownOptions(TRUE, FALSE);

    trace("\n\n## Start of source bottom-up and destination top-down tests.\n\n");
    test_StretchBlt_TopDownOptions(FALSE, TRUE);

    trace("\n\n## Start of source bottom-up and destination bottom-up tests.\n\n");
    test_StretchBlt_TopDownOptions(FALSE, FALSE);
```